com.unity.netcode.gameobjects/Runtime/SceneManagement/ScenesInBuild.cs

```diff
@@ -0,0 +1,255 @@
+                ( (networkManager.ScenesInBuild == null || networkManager.ScenesInBuild != null && networkManager.ScenesInBuild.Scenes.Count == 0) && EditorApplication.isPlaying) )
+            {
+                if (networkManager.ScenesInBuild == null)
+                {
+                    networkManager.ScenesInBuild = InitializeScenesInBuild(networkManager);
+                }
+
+                networkManager.ScenesInBuild.PopulateScenesInBuild();
+            }
+#else
+
+            // Only if we are running a stand alone build as for unit or integration testing
+            if (isUnitTestRunning)
+            {
+                if (networkManager.ScenesInBuild == null)
+                {
+                    networkManager.ScenesInBuild = CreateInstance<ScenesInBuild>();
+                }
+                var currentlyActiveSceneName = SceneManager.GetActiveScene().name;
+
+                // If the unit test scene is not already added to the ScenesInBuild
+                if(!networkManager.ScenesInBuild.Scenes.Contains(currentlyActiveSceneName))
+                {
+                    // Then add it into the valid scenes that it can be loaded
+                    networkManager.ScenesInBuild.Scenes.Add(currentlyActiveSceneName);
+                }
+            }
+#endif
+        }
+
+
+#if UNITY_EDITOR
+
+        private bool m_CheckHasBeenApplied;
+
+        /// <summary>
+        /// This will add a play mode state change watch to determine when we are done with a unit test
+        /// in order to re-synchronize our ScenesInBuild asset.
+        /// </summary>
+        private void CheckForEndOfUnitTest()
+        {
+            if (!m_CheckHasBeenApplied)
+            {
+                EditorApplication.playModeStateChanged += EditorApplication_playModeStateChanged;
+                m_CheckHasBeenApplied = true;
+            }
+        }
+
+        /// <summary>
+        /// Check for when we enter into editor mode so we can remove any scenes that might have been added to the
+        /// scenes list.
+        /// </summary>
+        private void EditorApplication_playModeStateChanged(PlayModeStateChange obj)
+        {
+            if (obj == PlayModeStateChange.EnteredEditMode)
+            {
+                EditorApplication.playModeStateChanged -= EditorApplication_playModeStateChanged;
+                m_CheckHasBeenApplied = false;
+                PopulateScenesInBuild();
+            }
+        }
+
+        /// <summary>
+        /// This will create a new ScenesInBuildList asset if one does not exist and will adjust the path to the ScenesInBuildList
+        /// asset if the asset is moved.  This will also notify the user if more than one ScenesInBuildList asset exists.
+        /// </summary>
+        /// <param name="networkManager">The relative network manager instance</param>
+        /// <returns></returns>
+        internal static ScenesInBuild InitializeScenesInBuild(NetworkManager networkManager)
```

Comment:
can we avoid relying on `ScenesInBuildList.asset`? can we access scene list at runtime in a standalone build and iterate over them to grab their indices instead?